service/src/main/java/tds/assessment/web/endpoints/ItemController.java

```diff
@@ -0,0 +1,74 @@
+package tds.assessment.web.endpoints;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+import tds.assessment.ItemConstraint;
+import tds.assessment.ItemProperty;
+import tds.assessment.services.ItemService;
+import tds.common.web.exceptions.NotFoundException;
+
+/**
+ * Controller handling item related entities
+ */
+@RestController
+@RequestMapping("/assessments/items")
+public class ItemController {
+    private final ItemService itemService;
+
+    @Autowired
+    public ItemController(ItemService itemService) {
+        this.itemService = itemService;
+    }
+
+    /**
+     * Endpoint to find the list of {@link tds.assessment.ItemConstraint}s for the assessment and client
+     *
+     * @param clientName    the client environment name
+     * @param assessmentId  the assessment id of the assessment that the constraints belong to
+     * @return {@link org.springframework.http.ResponseEntity} containing a list of  {@link tds.assessment.ItemConstraint}}
+     * @throws tds.common.web.exceptions.NotFoundException if no results are found
+     */
+    @RequestMapping(value = "/constraints/{clientName}/{assessmentId}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
I think the URL for this things should be `assessments/{assessmentid}/{client}/items/constraints` or `assessments/{client}/{assessmentid}/items/constraints`.

Also, shouldn't this be all contained in the assessment fetching the items?  I don't follow the lines that were drawn here with what we are fetching when we get an assessment vs this code.